### PR TITLE
use `gh issue comment` instead of peter-evans/create-or-update-comment

### LINKED ISSUE #29419: Use GitHub CLI in 'Commenting on an issue when a label is added' page instead of 3rd party action
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added

### What part(s) of the article would you like to see updated?

the use of https://github.com/marketplace/actions/create-or-update-comment, while cool, doesn't seem quite right. you're recommending a third-party action, adding a huge disclaimer about it being third-party, and pinning to a specific hash; all while there's a cool `gh issue comment` command that's **always installed** via the github-provided base runner images.

i think it'd be good to replace the recommendation to use the `gh` cli instead. this would remove the need for the large scary disclaimer and make it more "official" looking and associated more strongly with the github brand.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by jcbhmr**
### Why:

Closes: #29419

### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline.

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

File: content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md [hunk 1]
```diff
@@ -30,38 +30,38 @@ In the tutorial, you will first make a workflow file that uses the [`peter-evans
...
           issues: write
         steps:
           - name: Add comment
-            uses: peter-evans/create-or-update-comment@5f728c3dae25f329afbe34ee4d08eef25569d79f
-            with:
-              issue-number: {% raw %}${{ github.event.issue.number }}{% endraw %}
-              body: |
-                This issue is available for anyone to work on. **Make sure to reference this issue in your pull request.** :sparkles: Thank you for your contribution! :sparkles:
+            # https://cli.github.com/manual/gh_issue_comment
+            run: gh issue comment -R "$REPOSITORY" "$NUMBER" --body "$BODY"
```
Comment (peterbe) [inline content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md:47]: Seems a bit "excessive" to create an environment variable only to use it here, once, directly. 
What do you think about simplifying to 
```
run: gh issue comment ${{ github.event.issue.number }} --repo ${{ github.repository }} --body "$BODY"
```
Also, note the removal of quotation marks given that an issue number will not have a space in it. Neither will the `github.repository`. 

And the order when you run `gh issue comment --help` is this:
```
USAGE
  gh issue comment {<number> | <url>} [flags]
```

So, we mention the issue number first and then all flags together after.

File: content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md [hunk 2]
```diff
@@ -30,38 +30,38 @@ In the tutorial, you will first make a workflow file that uses the [`peter-evans
...
+            env:
+              GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (peterbe) [inline content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md:49]:
```suggestion
              GH_TOKEN: ${{ github.token }}
```

I know that works. Because [I tested it](https://github.com/peterbe/github-action-tricks/issues/53#issuecomment-1781171488) :)
But what I'm not sure of is if it's established best practice in documentation.

File: content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md [hunk 3]
```diff
@@ -30,38 +30,38 @@ In the tutorial, you will first make a workflow file that uses the [`peter-evans
 1. Copy the following YAML contents into your workflow file.
 
     ```yaml copy
-    {% data reusables.actions.actions-not-certified-by-github-comment %}
-
-    {% data reusables.actions.actions-use-sha-pinning-comment %}
-
```
Comment (peterbe) [inline content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md:36]: Why were these removed?

File: content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md [hunk 4]
```diff
@@ -30,38 +30,38 @@ In the tutorial, you will first make a workflow file that uses the [`peter-evans
...
         permissions:
           issues: write
         steps:
           - name: Add comment
-            uses: peter-evans/create-or-update-comment@5f728c3dae25f329afbe34ee4d08eef25569d79f
-            with:
-              issue-number: {% raw %}${{ github.event.issue.number }}{% endraw %}
-              body: |
-                This issue is available for anyone to work on. **Make sure to reference this issue in your pull request.** :sparkles: Thank you for your contribution! :sparkles:
+            # https://cli.github.com/manual/gh_issue_comment
```
Comment (peterbe) [inline content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md:46]: It's up to the Docs reviewers but I think this is a bit unnecessary given that you're on the documentation already. 
If you can use `gh issue ...` you, in a sense, don't need the manual.

File: content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md [hunk 5]
```diff
@@ -72,4 +72,4 @@ Test your workflow by applying your specified label to an issue.
 
 ## Next steps
 
-- To learn more about additional things you can do with the `peter-evans/create-or-update-comment` action, like adding reactions, visit the [`peter-evans/create-or-update-comment` action documentation](https://github.com/marketplace/actions/create-or-update-comment).
+- To learn more about additional things you can do with the GitHub CLI, like editing existing comments, visit the [GitHub CLI Manual](https://cli.github.com/manual/).
```
Comment (peterbe) [inline content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md:75]: 👍

**Review by peterbe — DISMISSED** [review]
I'm not a Docs content reviewer but I endorse the idea of switching the third-party action to instead using the `gh` CLI. 

I left some nits for the reviewer.

File: content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md (hunk 1 above)
Comment (jcbhmr) [reply content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md:47]: imo with all that one one line its hard to read, especially on a phone screen. so you're left with something like this:

```yml
run: |
  gh issue comment ${{ github.event.issue.number }} \
    --repo ${{ github.repository }} --body "$BODY"
```

idk. i just put it in env vars so the `run: ...` line was one line. it also had the added benefit of feeling more like a `with: body: ...` input on a normal action 🤷‍♂️

File: content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md (hunk 2 above)
Comment (jcbhmr) [reply content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md:48]: is there precedence in the docs one way or the other? 🤔 for users, secrets.GITHUB_TOKEN is 8x more popular than github.token from preliminary searching https://github.com/search?q=secrets.GITHUB_TOKEN&type=code https://github.com/search?q=github.token&type=code 🤷‍♀️

File: content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md (hunk 3 above)
Comment (jcbhmr) [reply content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md:36]: they add this boilerplate:

```yml
# This workflow uses actions that are not certified by GitHub.
# They are provided by a third-party and are governed by
# separate terms of service, privacy policy, and support
# documentation.

# GitHub recommends pinning actions to a commit SHA.
# To get a newer version, you will need to update the SHA.
# You can also reference a tag or branch, but the action may change without warning.
```

which is no longer needed since
1. we aren't using **any** third party actions anymore so don't need disclaimer
2. there's no actions tot apply the "pin your actions" recommendation to so don't need it

so i figured i should remove 'em lol

Comment (peterbe) [reply content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md:36]: > we aren't using any third party actions *anymore*

(emphasis mine)

That's a great point! Thanks for making this comment. It helps understanding the reasoning for deleting those. Good choice 👍

**Comment by jcbhmr** [comment]
i forgot to mention that this pr also happens to fix this bug:

```yml
if: github.event.label.name == 'help-wanted'
```
👆 doesn't work since the label name is "help wanted" not "help-wanted" and so it never triggers 😢

```yml
if: github.event.label.name == 'help wanted'
```
👆 this works ✅
its whatever name appears here:
image


so this one has "waiting for review" right now but NOT "waiting-for-review". you can check this in the search by clicking on it and seeing whats in the search box/query:
image

"waiting for review" NOT "waiting-for-review"

idk if there are other spots like this in other parts of the documentation too 🤷‍♀️

**Comment by smoothzebra** [comment]
ในวันที่ พฤ. 26 ต.ค. 2023 00:48 น. Jacob Hummer ***@***.***>
เขียนว่า:

File: content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md (hunk 2 above)
Comment (peterbe) [reply content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md]: I honestly don't know. I vaguely remember that `github.token` is new(er). 

I think it's for the @github/docs-content reviewer to try to find that out. It would certainly be good to have a one blessed choice given that they appear to accomplish the same thing. 

Personally, I find `github.token` preferred because it's clearly something I don't mess with. Whereas if I, as a user, create a secret called `SECRET_INGREDIENT` it would be `${{ secrets.SECRET_INGREDIENT }}` that I use. So it feels like the `secrets. ...` is the nomenclature for *your* custom secrets. 

It feels like it says: "You can create and manage your own secrets. But I've created one for you that you can't create and manage."

Comment (skedwards88) [reply content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md]: We typically recommend `secrets.GITHUB_TOKEN`

File: content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md (hunk 1 above)
Comment (skedwards88) [reply content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md]: We typically do try to use variables to avoid injection attacks, even though it is more overhead/more verbose. An injection attack is not really a risk here, but it gets people in the habit of coding defensively.

**Review by skedwards88 — COMMENTED** [review]
Thanks for this PR! I love this change. I left a few minor comments that I'll go ahead an incorporate, then we'll get this merged for you.

**Review by skedwards88 — DISMISSED** [review]

**Comment by Yme4u** [comment]
Thank you all I'm so beat tired. I need to sign up and get busy. Thanks . Good nite

Sent from Yahoo Mail on Android 
 
  On Fri, Oct 27, 2023 at 10:20 AM, ***@***.***> wrote:   


#29420 was automatically removed from the merge queue due to failed status checks.

File: content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md [hunk 6]
```diff
@@ -30,38 +30,38 @@ In the tutorial, you will first make a workflow file that uses the [`peter-evans
...
         permissions:
           issues: write
         steps:
           - name: Add comment
-            uses: peter-evans/create-or-update-comment@5f728c3dae25f329afbe34ee4d08eef25569d79f
-            with:
-              issue-number: {% raw %}${{ github.event.issue.number }}{% endraw %}
-              body: |
-                This issue is available for anyone to work on. **Make sure to reference this issue in your pull request.** :sparkles: Thank you for your contribution! :sparkles:
+            run: gh issue comment "$NUMBER" --repo "$REPO" --body "$BODY"
```
[the remark this answers is not in the document]
Comment (jcbhmr) [reply content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md]: actually the current iteration of this workflow **does not work** because you **need the `--repo` flag** since there's no `git remote get-url` context for the `gh` command to use to infer which github repo is the remote that it should target.

image

instead, it needs to have the `--repo` flag

**Comment by jcbhmr** [comment]
turns out the workflow _does_ need the `--repo` flag lol 🤣

image
image

vs with the `--repo` flag

image
image

i think its down to that `gh` needs a `.git` folder (current git repo context) to figure out which remote github repo it corresponds to. if there isn't one then i think you're always required to specify it manually via `-R`/`--repo`. 🤷‍♂️

File: content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (skedwards88) [reply content/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added.md:46]: 🤔 Interesting. Thanks for verifying and correcting that! I'll get this reapproved and merged for you.

**Review by skedwards88 — APPROVED** [review]
